### DISCUSSION TITLE
Using Wazuh with osquery

### DISCUSSION THREAD

**Discussion by Bilal-SG** [wazuh/wazuh#15964] [General]
Hello Wazuh Community!

I am searching for a way to configure Wazuh to run interactive osquery queries. I have used  Wazuh in managing osqueryd in agents side and configure it to send scheduled osquery results to the manager. My question is there a way to use Wazuh to open osquery interactive shell in the monitored agents and use it to send live queries to them?

Noting the Wazuh I have is composed of (Indexer + Manager + Dashboard).

Thanks in advance,
Bilal

**Comment by jctello** [comment]
Hi @Bilal-SG ,

Through [centralized configuration](https://documentation.wazuh.com/current/user-manual/agents/grouping-agents.html) you may distribute osquery configuration files and configure the agent to use these configuration files, for example by creating the `osquery` group, creating a `/var/ossec/etc/shared/osquery/osquery.conf` file containing the queries and modifying that group's `agent.conf` to be:
```
<agent_config>
<wodle name="osquery">
      <disabled>no</disabled>
      <run_daemon>yes</run_daemon>
      <bin_path>/usr/bin</bin_path>
      <log_path>/var/log/osquery/osqueryd.results.log</log_path>
      <config_path>/var/ossec/etc/shared/osquery.conf</config_path>
      <add_labels>yes</add_labels>
</wodle>
</agent_config>
```
This will allow you to control from the manager the queries performed on the agent by modifying the `osquery.conf` file.  Each time the centralized configuration of an agent is modified the agent restarts and will execute osquery with the new configuration.

It is important that the osquery service is not running as a daemon on the monitored endpoint so that it is only executed by the Wazuh agent.

An interactive shell is not a feature of the osquery wodle, and Wazuh is designed to prohibit the execution of arbitrary commands from the manager in order to limit the privileges of the platform over the monitored endpoints; However you may create a custom active-response to create an on-demand osquery shell to the monitored endpoints if you consider that the enhanced visibility justifies the additional privileges this would grant the manager.

**Comment by bremoi** [comment]
Hi,

Is it possible to run live osquery queries just for questioning and not necessarily for alerting? In other words, I would like to execute a query on all agents like FleetDM does.

Thanks a lot in advance.

**Reply by bremoi** [reply]
In addition, more information at https://github.com/wazuh/wazuh/discussions/15964.
